app/src/main/java/org/mozilla/fenix/perf/RunBlockingCounter.kt

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This class implements the alternative ways to invoke runBlocking with some
+// monitoring by wrapping the raw methods. This lint check tells us not to use the raw
+// methods so we suppress the check.
+@file:Suppress("MozillaRunBlockingCheck")
+
+package org.mozilla.fenix.perf
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Counts the number of runBlocking calls made
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+object RunBlockingCounter {
+    var count = AtomicInteger(0)
+}
+
+/**
+ * Wrapper around `runBlocking`. RunBlocking seems to be a "fix-all" to return values to the thread
+ * where the coroutine is called. The official doc explains runBlocking:  "Runs a new coroutine and
+ * blocks the current thread interruptibly until its completion`. This can have negative
+ * side-effects on the our main thread which could lead to significant jank. This wrapper aims to
+ * count the number of runBlocking call to try to limit them as much as possible to encourage
+ * alternatives solutions whenever this function might be needed.
+ */
+fun <T> runBlockingIncrement(
+    context: CoroutineContext? = null,
+    action: suspend CoroutineScope.() -> T
+): T {
+    synchronized(lock = Any()) {
```

Comment:
This won't synchronized properly: each time we go through this, we lock a new instance so there is never any contention for this code block. You need to use a shared instance.

That being said, do we know that this class throws an exception on overflow? Maybe it just does something safe.

If not, I think the synchronization block could be replaced with something like `getAndUpdate`: https://developer.android.com/reference/java/util/concurrent/atomic/AtomicInteger?hl=en#getAndUpdate(java.util.function.IntUnaryOperator)

Reply:
@mcomella  so `getAndUpdate` is min API 24. Also, the `AtomicInteger` seems to wrap around like a regular integer.  We could just have a function in the `object` and let it handle it itself?